### PR TITLE
Relayer discord bot

### PR DESCRIPTION
**Pull request by stefan-nikolov96**

### CODE REVIEW

File: relay/utils/discord_monitor.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    private async dispatchMessage(messageToSend) {
+        this.client.channels.cache.get(env.channel_id).send(messageToSend)
+    }
+
+    public async monitor_delay() {
+
+        for (var contract of Object.keys(this.contracts)) {
+            var name = contract;
+            var delay = await this.getSlotDelay(this.contracts[contract].SolidityContract!)
```
Comment (zah) [inline relay/utils/discord_monitor.ts:132]: Could this line hang forever? Is there a timeout somewhere underneath in the libraries making this web3 requests?

**Review by Dimo99 — COMMENTED** [review]
- Use `let`/`const` instead of `var` as var is generally not recommended to be used you can read more in this article or research the topic yourself: https://www.typescriptlang.org/docs/handbook/variable-declarations.html
- Use `import` instead of `require` as it is more idiomatic to typescript
- If possible give `client` a specific type instead of any
- Load the env variables with `process.env` object instead of relying on a specific env file path
- run `yarn format:write` to fix the linting problems

**Comment by EmilIvanichkovv** [comment]
Regarding Dimo's advice:
> Load the env variables with process.env object instead of relying on a specific env file path

Even better wrap it with the `assertNotNull`

File: relay/implementations/solidity-contract.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import {
 } from './publish_evm_transaction';
 
 export class SolidityContract implements ISmartContract {
-  private lightClientContract: Contract;
+  private lightClientContract: Contract;
```
Comment (EmilIvanichkovv) [inline relay/implementations/solidity-contract.ts:11]: Remove the empty space please

File: relay/utils/discord_monitor.ts (hunk 1 above)
Comment (stefan-nikolov96) [reply relay/utils/discord_monitor.ts:132]: The BeaconApi uses the native 'fetch' function from Node.js which doesn't implement timeouts. Ethers.js contracts don't seem to implement timeouts as well, so I will wrap the display_monitor() invocation with a Promise.race() with itself and a timeout and retry this statement 10 times before throwing an error.

File: relay/implementations/solidity-contract.ts (hunk 2 above)
Comment (stefan-nikolov96) [reply relay/implementations/solidity-contract.ts]: done

**Review by Dimo99 — APPROVED** [review]

**Pull request merged**